python/xorbits/deploy/kubernetes/client.py

```diff
@@ -0,0 +1,506 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import functools
+import logging
+import time
+import uuid
+from typing import Collection, Dict, List, Optional, Union
+
+from ..._mars.deploy.utils import wait_services_ready
+from ..._mars.lib.aio import new_isolation, stop_isolation
+from ..._mars.services.cluster.api import WebClusterAPI
+from ..._mars.session import new_session
+from ..._mars.utils import calc_size_by_str
+from .config import (
+    IngressConfig,
+    NamespaceConfig,
+    RoleBindingConfig,
+    RoleConfig,
+    ServiceConfig,
+    XorbitsSupervisorsConfig,
+    XorbitsWorkersConfig,
+)
+
+try:
+    from kubernetes.client import ApiClient
```

Comment:
We can consider to use `lazy_import` for this situation.

```python
from .._mars.utils import lazy_import

k8s_client = lazy_import('kubernetes.client')
k8s_rest = lazy_import('kubernetes.client.rest')
```

When we want to use the ApiClient in the function, we can just use `k8s_client.ApiClient`, if `kubernetes` is not installed, the error will be raised.

Reply:
Fix